File: app/src/main/java/maderski/bluetoothautoplaymusic/utils/BluetoothUtils.java
package maderski.bluetoothautoplaymusic.utils;

import android.bluetooth.BluetoothAdapter;
import android.bluetooth.BluetoothDevice;
import android.content.Context;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

import maderski.bluetoothautoplaymusic.R;

/**
 * Created by Jason on 1/14/17.
 */

public class BluetoothUtils {
    public static final String TAG = "BluetoothDeviceUtils";

    // List of bluetooth devices on the phone
    public static List<String> listOfBluetoothDevices(Context context) {
        List<String> btDevices = new ArrayList<String>();

        BluetoothAdapter bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();

        if(bluetoothAdapter != null) {
            Set<BluetoothDevice> pairedDevices = bluetoothAdapter.getBondedDevices();

            for(BluetoothDevice bt : pairedDevices)
                btDevices.add(bt.getName());
        }else{
            btDevices.add(0, context.getString(R.string.no_bluetooth_device));
        }

        return btDevices;
    }

    public static void enableDisabledBluetooth(){
        BluetoothAdapter bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
        if(bluetoothAdapter != null) {
            if (!bluetoothAdapter.isEnabled()) {
                bluetoothAdapter.enable();
            }
        }
    }
}

File: app/src/main/java/maderski/bluetoothautoplaymusic/utils/BluetoothUtils.kt
package maderski.bluetoothautoplaymusic.utils

import android.bluetooth.BluetoothAdapter
import android.bluetooth.BluetoothDevice
import android.content.Context
import android.media.AudioDeviceInfo
import android.media.AudioManager
import android.os.Build

import java.util.ArrayList

import maderski.bluetoothautoplaymusic.R

/**
 * Created by Jason on 1/14/17.
 */

object BluetoothUtils {
    const val TAG = "BluetoothDeviceUtils"

    // List of bluetooth devices on the phone
    fun listOfBluetoothDevices(context: Context): List<String> {
        val btDevices = ArrayList<String>()

        val bluetoothAdapter = BluetoothAdapter.getDefaultAdapter()

        if (bluetoothAdapter != null) {
            val pairedDevices = bluetoothAdapter.bondedDevices

            for (bt in pairedDevices)
                btDevices.add(bt.name)
        } else {
            btDevices.add(0, context.getString(R.string.no_bluetooth_device))
        }

        return btDevices
    }

    fun enableDisabledBluetooth() {
        val bluetoothAdapter = BluetoothAdapter.getDefaultAdapter()
        if (bluetoothAdapter != null) {
            if (!bluetoothAdapter.isEnabled) {
                bluetoothAdapter.enable()
            }
        }
    }

    fun isBluetoothA2DPOnCompat(context: Context): Boolean {
        val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
        return if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
            audioManager.isBluetoothA2dpOn
        } else {
            audioManager.getDevices(AudioDeviceInfo.TYPE_BLUETOOTH_A2DP).isNotEmpty()
        }
    }
}
